rpc/coretypes/responses_test.go

```diff
@@ -33,3 +41,45 @@ func TestStatusIndexer(t *testing.T) {
 		assert.Equal(t, tc.expected, status.TxIndexEnabled())
 	}
 }
+
+// A regression test for https://github.com/tendermint/tendermint/issues/8583.
+func TestResultBlockResults_regression8583(t *testing.T) {
+	const keyData = "0123456789abcdef0123456789abcdef" // 32 bytes
+	wantKey := base64.StdEncoding.EncodeToString([]byte(keyData))
+
+	rsp := &ResultBlockResults{
+		ValidatorUpdates: []abci.ValidatorUpdate{{
+			PubKey: pbcrypto.PublicKey{
+				Sum: &pbcrypto.PublicKey_Ed25519{Ed25519: []byte(keyData)},
+			},
+			Power: 400,
+		}},
+	}
+
+	// Use compact here so the test data remain legible. The output from the
+	// marshaler will have whitespace folded out so we need to do that too for
+	// the comparison to be valid.
+	var buf bytes.Buffer
+	require.NoError(t, json.Compact(&buf, []byte(fmt.Sprintf(`
+{
+  "height": "0",
+  "txs_results": null,
+  "total_gas_used": "0",
+  "finalize_block_events": null,
+  "validator_updates": [
+    {
+      "pub_key":{"type": "tendermint/PubKeyEd25519", "value": "%s"},
+      "power": "400"
+    }
+  ],
+  "consensus_param_updates": null
+}`, wantKey))))
+
+	bits, err := json.Marshal(rsp)
+	if err != nil {
+		t.Fatalf("Encoding block result: %v", err)
+	}
+	if diff := cmp.Diff(buf.String(), string(bits)); diff != "" {
+		t.Errorf("Marshaled result (-want, +got):\n%s", diff)
+	}
```

Comment:
Why this and not something like require.Equal(....)

Reply:
Because the output from `require.Equal` doesn't provide a meaningful diff, and most of the test input is not directly relevant to the bug.